src/osd/OSD.cc

```diff
@@ -9634,10 +9634,12 @@ void OSD::dequeue_op(
 	   << " pg " << *pg << dendl;
 
   logger->tinc(l_osd_op_before_dequeue_op_lat, latency);
-
-  service.maybe_share_map(m->get_connection().get(),
-			  pg->get_osdmap(),
-			  op->sent_epoch);
+  // in send_message_osd_cluster will call mabye_share_map.
+  if (!m->get_source().is_osd()) {
+    service.maybe_share_map(m->get_connection().get(),
```

Comment:
hmm, actually I'd prefer a unique handling logic (more maintainable and less error-prone) over the performance benefit here, which is obviously small...

Reply:
Thanks! But i think this path is the critical path. So it is worth doing this.

Reply:
I am leaving it to Neha. @majianpeng

Reply:
I agree with @xiexingguo's concerns here. Perhaps, we can run this through some testing to see how it does?